srtcore/queue.h

```diff
@@ -67,6 +67,36 @@ namespace srt
 {
 class CChannel;
 class CUDT;
+class CUDTWrapper;
+
+class CUDTWrapper {
+public:
+    CUDT *udt;
+    sync::SharedMutex mut;
+
+public:
+    CUDTWrapper()
+    :udt(NULL)
+    ,mut()
+    {
+}
+void lockRead()
+{
+    return mut.lockRead();
+}
+void lockWrite()
+{
+    return mut.lockWrite();
+}
+void unlockRead()
+{
+    return mut.unlockRead();
+
+}
+void unlockWrite(){
+    return mut.unlockWrite();
+}
+};
```

Comment:
I don't quite like the name. It does not express what the class does, except for wrapping `CUDT`.
In fact, it can be a template class holding a pointer to a resource and a shared mutex, e.g.

```c++
template <class T>
class CSharedObject
{
private:
    T* m_pObj;
    sync::SharedMutex m_mtx;
public:
    // ...
};
```